editor/export/editor_export_platform.cpp

```diff
@@ -822,7 +822,7 @@ String EditorExportPlatform::_export_customize(const String &p_path, LocalVector
 	if (type == "PackedScene") { // Its a scene.
 		Ref<PackedScene> ps = ResourceLoader::load(p_path, "PackedScene", ResourceFormatLoader::CACHE_MODE_IGNORE);
 		ERR_FAIL_COND_V(ps.is_null(), p_path);
-		Node *node = ps->instantiate(PackedScene::GEN_EDIT_STATE_INSTANCE); // Make sure the child scene root gets the correct inheritance chain.
+		Node *node = ps->instantiate(PackedScene::GEN_EDIT_STATE_MAIN); // Make sure scene root gets the correct inheritance chain.
```

Comment:
My understanding of the code (which could be wrong) is that this line is calling instantiate on every packed scene in the game as part of some necessary processing when exporting the game to a binary file. Since we are just processing the scenes alone by themselves, it makes sense to use `GEN_EDIT_STATE_MAIN` instead of `INSTANCE` here. 

I believe passing `GEN_EDIT_STATE_INSTANCE` to `instantiate` was applying the `CONNECT_INHERITED` flag falsely to signal connections and saving them. It was not game breaking, so went unnoticed. But by changing this, we can use the `CONNECT_INHERITED` flag to correctly process signals when packing a scene. 

But this should be verified by someone more familiar with the engine.

Reply:
It's hard to be 100% sure about the implications without detailedly debugging all this myself, as it's long since I last touched this code, but I'd say your logic is good.

The use of `GEN_EDIT_STATE_INSTANCE` was in fact discussed (see https://github.com/godotengine/godot/commit/6973abd458ff8f13837b1e44857107e34dcd34c7). The reason stated there for prefering it was a bit weak. and there wasn't a strong preference for either. Now we have a solid argument to back why `GEN_EDIT_STATE_MAIN` is preferable. I agree with you that it's the right choice in this context, where the scene is being instantiated in a standalone fashion, which warrants it to be treated as the main one.

Reply:
I changed the implementation and reverted this change since it is no longer needed.